### PR TITLE
chore(aws-eks): update README with correct service account syntax

### PR DESCRIPTION
**Pull request by pahud**
chore(aws-eks): update README with correct service account syntax

`serviceAccountName` should be at `spec.serviceAccountName` and add a CfnOutput sample to demo how to get the IAM role of this service account.

My working sample here
https://twitter.com/pahudnet/status/1263286407092514817

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by pahud** [comment]
consider this scenario

```ts
    const sa = cluster.addServiceAccount('MyServiceAccount', {});

    const pod = cluster.addResource('mypod2', {
      apiVersion: 'v1',
      kind: 'Pod',
      metadata: { name: 'mypod2' },
      spec: {
        serviceAccountName: sa.serviceAccountName,
        containers: [
          {
            name: 'main',
            image: 'pahud/aws-whoami',
            ports: [{ containerPort: 5000 }],
          }
        ]
      }
    });
```

It looks like the pod will not wait until the serviceAccount is read and hence end up with this error

圖片


addDependency() will fix this issue.

```ts
pod.node.addDependency(sa);
```

**Comment by vlesierse** [comment]
This is not something I ran into yet, but the situation seems to be valid when the resource is created before the service account. Think we would need to update the README to highlight this unless CDK can figure out the order based on the sa.serviceAccountName reference.

**Comment by pahud** [comment]
> This is not something I ran into yet, but the situation seems to be valid when the resource is created before the service account. Think we would need to update the README to highlight this unless CDK can figure out the order based on the sa.serviceAccountName reference.

OK noted in the README

File: packages/@aws-cdk/aws-eks/README.md [hunk 1]
```diff
@@ -526,26 +526,33 @@ With services account you can provide Kubernetes Pods access to AWS resources.
 
 ```ts
 // add service account
-const serviceAccount = cluster.addServiceAccount('MyServiceAccount');
+const sa = cluster.addServiceAccount('MyServiceAccount', {});
```
Comment (eladb) [inline packages/@aws-cdk/aws-eks/README.md:529]: The `{}` should not be needed. If it is, it's a bug.

```suggestion
const sa = cluster.addServiceAccount('MyServiceAccount');
```

Comment (vlesierse) [reply packages/@aws-cdk/aws-eks/README.md:529]: This has been fixed with in PR #8084 and should be removed.

Comment (pahud) [reply packages/@aws-cdk/aws-eks/README.md]: OK. Removed.

**Pull request merged**